src/main/java/com/box/sdk/BinaryBodyUtils.java

```diff
@@ -73,12 +74,36 @@ static void writeStreamWithContentLength(BoxAPIResponse response, OutputStream o
             } else {
                 input = response.getBody();
             }
-            writeStreamTo(input, output, response.getContentLength());
+            writeStreamTo(input, output, getContentLengthFromAPIResponse(response));
         } finally {
             response.close();
         }
     }
 
+    /**
+     * Get the content length from the API response.
+     * In some cases, the Content-Length is not provided in the response headers.
+     * This could happen when getting the content representation for a compressed data.
+     * In that case the API will switch to chunk mode and provide the length in the "X-Original-Content-Length" header.
+     *
+     * @param response API response.
+     * @return Content length.
+     */
+    private static long getContentLengthFromAPIResponse(BoxAPIResponse response) {
+        long length = response.getContentLength();
+        try {
+            if (length == -1 && response.getHeaders().containsKey(X_ORIGINAL_CONTENT_LENGTH)) {
```

Comment:
The code on line 93 checks for the size of the body, could you add one more checked please here 
for the header response.getHeaders().containsKey(HttpHeaders.CONTENT_LENGTH) as well please. Basically, 
1. Check for response.getContentLength()
2. Check for response.getHeaders().containsKey(HttpHeaders.CONTENT_LENGTH)
3. Check for response.getHeaders().containsKey(X_ORIGINAL_CONTENT_LENGTH)

Reply:
Done